File: ClientApp/src/app/features/offer/models/offer.ts
export class Offer {
    id: number;
    title: string;
    description: string;
    startDateTime: Date;
    endDateTime: Date;
    isRespawningOffer: boolean;
    categoryId?: number;
    categoryName?: string | null;
    respawnStartTime?: number | null;
}

export class OfferCalendarDTO {
  id: number;
  title: string;
  start: Date;
  end: Date;
}

File: AngularDemo/AngularDemo/Models/Offer.cs
using System;

namespace AngularDemo.Models
{
    public class Offer
    {
        public int Id { get; set; }
		public string Title { get; set; }
		public string Description { get; set; }
        public DateTime StartDateTime { get; set; }
        public DateTime EndDateTime { get; set; }
		public bool IsRespawningOffer { get; set; }
        public int CategoryId { get; set; }
        public string CategoryName { get; set; }
        public DateTime? RespawnStartTime { get; set; }
    }
}
